airo-typing/airo_typing/__init__.py

```diff
@@ -99,13 +105,25 @@
 """4x4 camera extrinsic matrix,
 this is the homogeneous matrix that describes the camera pose in the world frame"""
 
+##########################
+# 3D and point cloud types
+##########################
 
-PointCloudType = Vector3DArrayType
-""" a (N,3) numpy array that represents a point cloud"""
+BoundingBox3DType = Tuple[Tuple[float, float, float], Tuple[float, float, float]]
+""" a tuple of two tuples that represent the min and max corners of a 3D bounding box"""
 
-ColoredPointCloudType = np.ndarray
-" an (N,6) numpy array that represents a point cloud with color information. Color is in RGB, float (0-1) format."
-# manipulator types
+PointCloudPositionsType = Vector3DArrayType
+""" a (N,3) float32 numpy array that represents a point cloud"""
 
-JointConfigurationType = np.ndarray
-"""an (N,) numpy array that represents the joint angles for a robot"""
+PointCloudColorsType = np.ndarray
+""" a (N,3) uint8 numpy array that represents the RGB colors of a point cloud"""
+
+PointCloudAttributesType = Dict[str, np.ndarray]
+""" a dictionary of numpy arrays that represent additional attributes of a point cloud, e.g. normals, confidence, etc. """
+
+
+@dataclass
```

Comment:
makes me wonder if we should rename this package to `airo-core` to facilitate putting both types, interfaces and some data containers here. This would be quite the breaking change though

Reply:
but I don't 100% like the idea of also storing data classes in a 'typing' package

Reply:
I see your point, but moving e.g. the Camera interfaces or SE3Container here feels a bit weird.